timesketch/api/v1/resources/upload.py

```diff
@@ -40,9 +41,75 @@
 class UploadFileResource(resources.ResourceMixin, Resource):
     """Resource that processes uploaded files."""
 
+    def _get_index(
+            self, name, description, sketch, index_name='',
+            label='', extension=''):
+        """Returns a SearchIndex object to be used for uploads.
+
+        Args:
+            name: the name of the searchindex.
+            description: the description of the searchindex.
+            sketch: sketch object (instance of Sketch).
+            index_name: optional index name, if supplied and if it exists
+                then the index associated with that will be returned.
+            label: optional label of the data, if supplied will be used to
+                determine whether an already existing index can be
+                used or a new one created.
+            extension: optional file extension if a file is being uploaded,
+                if supplied and no label used, then the extension will be
+                used as a label.
+
+        Returns:
+            A SearchIndex object.
+        """
+        if index_name:
+            if not isinstance(index_name, six.text_type):
+                index_name = codecs.decode(index_name, 'utf-8')
+
+            searchindex = SearchIndex.query.filter_by(
+                name=name, index_name=index_name).first()
+
+            if searchindex and searchindex.has_permission(
+                    permission='write', user=current_user):
+                return searchindex
+
+        if extension and not label:
+            label = extension
+
+        if not label:
+            label = 'log'
+
+        timelines = Timeline.query.filter_by(
```

Comment:
use sketch.active_timelines here to filter out processing, failed and archived timelines.

Reply:
ack